### ISSUE TITLE
License of the package - consider move to GPL-3+

### ISSUE THREAD

**Issue by matsl** [ProofGeneral/PG#198]
In a thread in the emacs-devel maillist, the licensing situation for emacs packages provided through Emacs package archives has been under focus. I have volunteered to contact the authors of packages that have a license that is incompatible with Emacs, which is now under GPL-3+.

See https://lists.gnu.org/archive/html/emacs-devel/2017-07/msg01069.html

So I wonder if you could consider to change the license of your package to GPL-3+?

Looking at the files there seems to be a mix of files with different license. The top package is under GPL-2 but there are files and sub modules that are distributed under GPL-3+.

**Comment by cpitclaudel** [comment]
We're working on it :) We've been working for most of the last 2 years, in fact :P And I think we're very close.

**Comment by matsl** [comment]
:+1:

**Comment by tarsius** [comment]
Any progress on this?

**Comment by erikmd** [comment]
@tarsius @matsl
FYI the related PR #195 has just been merged via commit 96aba0bfee1203b7b46d481f30410546ead68e09 in branch `async` (this branch gathering the new infrastructure of ProofGeneral for async Coq proofs that will be released in the next version of ProofGeneral)

**Comment by tarsius** [comment]
Thanks!

**Comment by tarsius** [comment]
Note to self: Mirror `master` again once `async` has been merged.

**Comment by tarsius** [comment]
I am now mirroring the `master` branch again and renamed the package to `proof-general`, using the same name as Melpa.

Currently this issue is resolved, but there is another issue and taking care of that *incompletely* would cause this issue to resurface, so I am raising that here:

The Emacsmirror needs a package to have a "main library", from which it extracts some metadata. Usually this is something like `foo.el` or `lisp/foo.el` for a package named `foo`. For `proof-general` I am currently using `pg-init.el`.

I was going to use `coq/coq.el`, but because that doesn't contain the proper permission statement just

    ;; License:     GPL (GNU GENERAL PUBLIC LICENSE)

my tools started claiming that the license is `GPL` when it should be `GPLv3+`. This could of course be fixed by putting the permission statement in that file too. But that's not quite enough.

First of all, what do *you* consider the "main file"?

I would like to encourage you to add `proof-general.el` even if that consists nothing but a header and `(require 'the-real-main-file)`. As mentioned above most packages come with a library that matches its name. The Emacsmirror and Melpa both get some information from that file. If it is missing then some metadata will be missing (see http://www.melpa.org/#/proof-general).

So please consider adding that file and
- adding the permission statement
- adding the Commentary section, its content will appear at http://www.melpa.org/#/proof-general, replacing `No description available.`.
- requiring either `coq/coq.el`, `pg-init.el` or if there is is something that makes more sense, that.

**Comment by cpitclaudel** [comment]
Partial answer: https://github.com/ProofGeneral/PG/issues/385 ; were planning on renaming pg-init to proof-general.el

**Comment by erikmd** [comment]
@tarsius Thanks a lot for your advice and feedback!

The state of this issue actually amounts to two points:

1. Regarding the license, we only moved to GPLv3+ in the `async` branch, so we did not close this issue #198 yet. Admittedly it would be much simpler to move to GPLv3+ in `master` as well (in order to be able to merge `async` and its PG/xml protocol into `master` at some point).
2. Regarding the "main file", both `pg-init.el` and `proof-site.el` could take this role. Actually the file `pg-init.el` could be preferred as it is a top-level file. As mentioned by @cpitclaudel we indeed plan to rename `pg-init.el` to `proof-general.el` (cf. #385). This will also let us address the Commentary/`No description available` issue you raised.

Thanks!
